### PR TITLE
Create logger Unit Tests

### PR DESCRIPTION
**Pull request by maysunfaisal**
Signed-off-by: Maysun J Faisal <maysun.j.faisal@ibm.com>

Issue #47

This PR:
- Creates Unit Tests for Turbine logger module

```
  22 passing (22s)

-----------------------------|----------|----------|----------|----------|-------------------|
File                         |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
-----------------------------|----------|----------|----------|----------|-------------------|
All files                    |    17.13 |     6.03 |    10.03 |    17.03 |                   |
 src                         |        0 |        0 |        0 |        0 |                   |
  index.ts                   |        0 |        0 |        0 |        0 |... 69,670,672,677 |
 src/controllers             |    12.97 |      2.9 |    14.13 |    13.09 |                   |
  projectEventsController.ts |    14.09 |     4.05 |        0 |    14.19 |... 69,270,271,273 |
  projectStatusController.ts |     16.9 |     2.67 |    25.93 |    17.08 |... 39,640,641,642 |
  projectsController.ts      |    10.65 |     2.77 |    10.53 |    10.75 |... 1151,1152,1153 |
 src/extensions              |    24.44 |        0 |        0 |       25 |                   |
  projectExtensions.ts       |    24.44 |        0 |        0 |       25 |... 34,246,247,250 |
 src/projects                |    15.14 |     4.81 |     6.09 |    14.69 |                   |
  DockerProject.ts           |    24.14 |      100 |        0 |    24.14 |... 50,162,163,165 |
  LibertyValidator.ts        |     6.82 |        0 |        0 |     6.82 |... 43,244,252,254 |
  Project.ts                 |    41.18 |       25 |       25 |    41.18 |... 41,144,154,155 |
  ShellExtensionProject.ts   |    11.59 |        0 |        0 |    11.59 |... 00,301,303,306 |
  SpringValidator.ts         |    15.79 |        0 |        0 |    16.22 |... 87,90,91,92,94 |
  Validator.ts               |    18.42 |       25 |    33.33 |    19.18 |... 20,221,222,223 |
  actions.ts                 |    13.97 |        0 |        0 |    13.97 |... 23,328,330,332 |
  constants.ts               |      100 |       90 |      100 |      100 |             13,16 |
  libertyProject.ts          |    31.09 |     3.45 |        0 |    31.62 |... 51,361,362,364 |
  logHelper.ts               |    30.49 |        0 |        0 |    31.65 |... 22,235,239,240 |
  nodejsProject.ts           |    47.54 |     8.33 |        0 |    47.54 |... 40,241,243,256 |
  operation.ts               |       25 |      100 |        0 |       25 | 21,22,23,34,35,36 |
  projectSpecifications.ts   |     7.97 |        0 |        0 |     8.03 |... 61,763,767,768 |
  projectUtil.ts             |     6.97 |     1.67 |     6.25 |     6.98 |... 2014,2017,2018 |
  springProject.ts           |    46.97 |     12.5 |        0 |    47.69 |... 26,227,229,241 |
  swiftProject.ts            |    59.26 |       50 |        0 |    59.26 |... 22,123,135,136 |
 src/utils                   |    26.42 |    16.25 |    16.88 |     27.1 |                   |
  dockerutil.ts              |     8.15 |        0 |        0 |     8.46 |... 16,617,618,620 |
  kubeutil.ts                |    13.18 |     5.77 |        0 |     13.6 |... 22,323,324,326 |
  locale.ts                  |    18.28 |     3.33 |        0 |    18.48 |... 19,220,223,224 |
  logger.ts                  |    85.71 |    66.13 |      100 |    89.73 |... 40,343,367,387 |
  processManager.ts          |    11.94 |        0 |        0 |    12.31 |... 82,183,184,187 |
  socket.ts                  |       40 |      100 |        0 |    44.44 |    33,34,35,43,44 |
  utils.ts                   |    37.25 |        0 |    16.67 |    37.25 |... 32,134,135,136 |
  workspaceSettings.ts       |    12.96 |        0 |        0 |    12.96 |... 75,280,281,283 |
-----------------------------|----------|----------|----------|----------|-------------------|
```

### CODE REVIEW

**Comment by maysunfaisal** [comment]
@rajivnathan @yangcao77 Can you pls help review the PR? Thx!

File: src/pfe/file-watcher/server/test/unit-test/tests/logger.module.test.ts [hunk 1]
```diff
@@ -0,0 +1,547 @@
...
+                await mkdirAsync(projectMetadataPath);
+                await copyAsync(originalProjectMetadata, testProjectMetadata);
+            }
+
+            process.env.NODE_ENV = "production";
+            logger.setLoggingLevel("debug");
+        });
+
+        after("remove test directories", async () => {
+            if ((await existsAsync(projectMetadataPath))) {
```
Comment (yangcao77) [inline src/pfe/file-watcher/server/test/unit-test/tests/logger.module.test.ts]: same as above

File: src/pfe/file-watcher/server/test/unit-test/tests/logger.module.test.ts [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+                await mkdirAsync(projectMetadataPath);
+                await copyAsync(originalProjectMetadata, testProjectMetadata);
+            }
+
+            process.env.NODE_ENV = "production";
+            logger.setLoggingLevel("trace");
+        });
+
+        after("remove test directories", async () => {
+            if ((await existsAsync(projectMetadataPath))) {
```
Comment (yangcao77) [inline src/pfe/file-watcher/server/test/unit-test/tests/logger.module.test.ts]: same as above

File: src/pfe/file-watcher/server/test/unit-test/tests/logger.module.test.ts [hunk 3]
```diff
@@ -0,0 +1,547 @@
...
+                await mkdirAsync(projectMetadataPath);
+                await copyAsync(originalProjectMetadata, testProjectMetadata);
+            }
+
+            process.env.NODE_ENV = "production";
+            logger.setLoggingLevel("info");
+        });
+
+        after("remove test directories", async () => {
+            if ((await existsAsync(projectMetadataPath))) {
```
Comment (yangcao77) [inline src/pfe/file-watcher/server/test/unit-test/tests/logger.module.test.ts]: same as above

File: src/pfe/file-watcher/server/test/unit-test/tests/logger.module.test.ts [hunk 4]
```diff
@@ -0,0 +1,547 @@
...
+            }).timeout(10000);
+        }
+    });
+
+    describe("testing of Turbine Log functions", () => {
+
+        before("set env variable", async () => {
+            process.env.NODE_ENV = "production";
+        });
+
```
Comment (yangcao77) [inline src/pfe/file-watcher/server/test/unit-test/tests/logger.module.test.ts]: same as above

File: src/pfe/file-watcher/server/test/unit-test/tests/projectExtensions.module.test.ts [hunk 5]
```diff
@@ -358,6 +364,12 @@ export function projectExtensionsTestModule(): void {
             const expectedResultProjectCapabilities = combinations[combo]["resultProjectCapabilities"];
 
             it(combo + " => getProjectHandler & getProjectCapabilities: projectID " + projectID, async () => {
+                if (combo == "combo6") {
+                    expect(fs.statSync(extensionsPath)).to.exist;
```
Comment (yangcao77) [inline src/pfe/file-watcher/server/test/unit-test/tests/projectExtensions.module.test.ts]: The existence of the required files/dirs has already checked  in before block. No need to recheck

**Comment by maysunfaisal** [comment]
@yangcao77 updated PR with feedback, thx

**Comment by yangcao77** [comment]
Changes look good to me.

**Pull request merged**